src/features/device/deviceGuides.tsx

```diff
@@ -67,6 +67,7 @@ const firmware = z.object({
         z.literal('Network'),
     ]),
     file: z.string().refine(f => ensureResourceExists(f)),
+    sampleSource: z.string().optional(),
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏Given that programming happens atomically for all the firmware files involved in a "choice" and we can therefore realistically only use one reference in the VS Code Step, I would argue that `sampleSource` is better housed in the object definition of `choice`, below.
This would also simplify the slightly-more-complex-than-necessary getter in `OpenVSCode.tsx`.

Reply:
Good idea. I change that.